### PR TITLE
Embedded authentication to resource functions. 

### PR DESCRIPTION
**Pull request by yashodgayashan**

### CODE REVIEW

File: core/src/env_platform/db_handler.bal [hunk 1]
```diff
@@ -291,9 +291,9 @@ function saveApplicationMetadata(string applicationType) returns boolean|error {
 # + return - This function will return whether the ministry is assigned or error if any occurs.
 function assignMinistry(AssignedMinistry assignedMinistry, string applicationId) returns boolean|error {
 
-    map<json>[] find = check applicationCollection->find({"applicationId": applicationId});
+    map<json>[] find = check applicationCollection->find({"applicationId": applicationId, status: "submit"});
 
-    // If no application is found.
+    // If no application is found or application is still a draft.
```
Comment (deshankoswatte) [inline core/src/env_platform/db_handler.bal:296]:
```suggestion
    // If no application is found or if the application is still a draft.
```

File: core/src/env_platform/db_handler.bal [hunk 2]
```diff
@@ -349,6 +349,8 @@ function isMinistry(string ministryId) returns boolean|error {
 # + return - This function will return either the updated status or appropriate error.
 function updateStatus(Status status, string applicationId) returns boolean|error {
     map<json>[] applications = check applicationCollection->find({"applicationId": applicationId, status: "submit"});
+
+    // If no application is found or application is still a draft.
```
Comment (deshankoswatte) [inline core/src/env_platform/db_handler.bal:353]:
```suggestion
    // If no application is found or if the application is still a draft.
```

**Comment by yashodgayashan** [comment]
@deshankoswatte resolved requested changes.

**Comment by deshankoswatte** [comment]
@sanjulamadurapperuma, @vikumkbv please review this PR.

File: core/src/env_platform/db_handler.bal [hunk 3]
```diff
@@ -349,6 +349,8 @@ function isMinistry(string ministryId) returns boolean|error {
 # + return - This function will return either the updated status or appropriate error.
 function updateStatus(Status status, string applicationId) returns boolean|error {
     map<json>[] applications = check applicationCollection->find({"applicationId": applicationId, status: "submit"});
+
+    // If no application is found or if the application is still a draft.
     if (applications.length() == 0) {
         return error("Not found", message = "Application with application Id: " + applicationId + " is not found.");
```
Comment (sanjulamadurapperuma) [inline core/src/env_platform/db_handler.bal:355]:
```suggestion
        return error("Not found", message = "Application with application ID: " + applicationId + " is not found.");
```

**Pull request merged**